### PR TITLE
Add labels to various objects

### LINKED ISSUE #800: Add label for amenity=car_sharing
amenity=car_sharing is currently rendered with icon, but without label. In bigger places there are multiple operators, so rendering the name or operator value would be nice.

See also https://trac.openstreetmap.org/ticket/3526.

### LINKED ISSUE #801: Add label for natural=tree
Tree names should be rendered. Some trees are important landmarks. Example: http://www.openstreetmap.org/#map=19/51.15624/6.81293

See also https://trac.openstreetmap.org/ticket/3267.

### LINKED ISSUE #840: Add rendering for post office label
Names of post offices are currently not rendered, but they probably should be.

See also https://trac.openstreetmap.org/ticket/5109.

### LINKED ISSUE #1091: Potential missing labels in amenity-points
Following objects are rendered as nodes on z20, but without labels for name tag

amenity=atm
amenity=bicycle_rental
highway=ford
tourism=information
lock=yes
man_made=mast
amenity=post_box
amenity=recycling
amenity=telephone
amenity=emergency_phone
amenity=toilets
amenity=drinking_water
man_made=water_tower
tourism=picnic_site
leisure=picnic_table
leisure=slipway
tourism=caravan_site

I think that man_made=mast and highway=ford may have labels, I am less sure about other positions.

In addition #459 is ticket for tourism=viewpoint, #840 amenity=post_office, #800 amenity=car_sharing

related to #710

### PR DESCRIPTION
**Pull request by math1985**
Add labels for:
* amenity=post_office (resolves #840)
* natural=tree (resolves #801)
* amenity=car_sharing (operator, resolves #800)
* amenity=atm (operator)
* amenity=bicycle_rental
* man_made=mast
* amenity=recycling
* amenity=drinking_water
* man_made=water_tower
* tourism=picnic_site
* leisure=picnic_table
* leisure=slipway

Resolves #1091.

### CODE REVIEW

**Comment by HolgerJeromin** [comment]
The car sharing thing is 800 instead of 880

**Comment by math1985** [comment]
> The car sharing thing is issue #800 instead of 880.

Thanks, corrected.

**Comment by matkoniecz** [comment]
There are text-dy problems with tall letters like É.

I tested post office - a is displayed but É is not.

amenity post_office name a operator b closed_way master - math_labels
amenity post_office name eeeeee eeeeee eeeeee ref 1 ele 8000 operator operator ee ee ee ee ee ee closed_way master - math_labels

**Comment by sb12** [comment]
Why don't you use brand for amenity=car_sharing and amenity=atm? This is the name the amenity is known by people and is often different to the actual operator. 

Examples:
Flinkster car sharing by the Deutsche Bahn: brand is Flinkster, while the actual operator is DB Rent.
Atms of the German Sparkasse are operated by the local Sparkasse branch e.g. Sparkasse Berlin while the brand is Sparkasse.

Some tests in Germany showed that often the operator and brand is not set correctly. But rendering the operator instead of the brand would even encourage this wrong tagging.

If the brand is not set we could still render the operator.

**Comment by math1985** [comment]
> Why don't you use brand for amenity=car_sharing and amenity=atm? This is the name the amenity is known by people and is often different to the actual operator. 

Brand is hardly ever used (for amenity=atm, operator is used 20 times as much as brand).

**Comment by math1985** [comment]
> It seems that there are text-dy problems with tall letters like É.

Strange, I thought I had tested everything. Updated.

**Comment by matkoniecz** [comment]
@math1985 Can you also rebase it?

**Comment by math1985** [comment]
Rebased.

**Comment by matkoniecz** [comment]
natural=tree labels are render for closed ways. I think that natural=tree labels are too small.

https://cloud.githubusercontent.com/assets/899988/6880483/0949192c-d534-11e4-94c5-e4c276774613.png

https://cloud.githubusercontent.com/assets/899988/6880484/0b15ff4a-d534-11e4-8c0d-1cfa28dc8c58.png

É fails for man_made=mast, amenity=recycling and amenity=dinking_water.

https://cloud.githubusercontent.com/assets/899988/6880490/466c443c-d534-11e4-855a-f78deb143e4c.png

leisure=slipway labels are render for closed ways.

File: amenity-points.mss [hunk 1]
```diff
@@ -1924,6 +1952,20 @@
     text-wrap-width: 30;
   }
 
+  [feature = 'natural_tree'][zoom >= 17] {
+    text-name: "[name]";
+    text-size: 8;
```
Comment (pnorman) [inline amenity-points.mss]: 8 is too small to be legible.

**Comment by math1985** [comment]
Rebased.

Increased font size for trees.

Dropped tree and slipway labels on areas.

**Comment by math1985** [comment]
> É fails for man_made=mast, amenity=recycling and amenity=drinking_water.

Fixed. Apparently whether a label fits does not only depend on text-dy and the icon, but also at where exactly the icon is rendered (which influences the antialias, I suppose). This makes testing this really tricky.

**Comment by math1985** [comment]
This is not ready for review.

**Comment by math1985** [comment]
This is now ready for review.

**Comment by matkoniecz** [comment]
I started review.

**Comment by matkoniecz** [comment]
I thought about different label for private amenity=recycling but that would be probably an overkill.

**Comment by matkoniecz** [comment]
[name=ÉÉÉÉÉÉ ÉÉÉÉÉÉ ÉÉÉÉÉÉ] is not displayed for man_made=mast

man_made mast name eeeeee eeeeee eeeeee ref 1 ele 8000 operator ee ee ee operator ee ee ee brand ee ee ee brand ee ee ee ee node master - math_labels

**Comment by math1985** [comment]
Thanks. Fixed and rebased.

**Comment by matkoniecz** [comment]
Note - this PR changes also label for amenity=bank. Larger text looks better, so there is no reason to revert this.

**Comment by matkoniecz** [comment]
Some banks with RTL names are losing labels, but that is result of #1313 (an upstream issue).

**Comment by matkoniecz** [comment]
@math1985 Thanks for labelification!

Squashed & merged in https://github.com/gravitystorm/openstreetmap-carto/commit/42cd70e6dba515478fd5a095cc837d5e03c86fe4

**Pull request closed without merging**